### PR TITLE
feat: set up repository

### PR DESCRIPTION
**Pull request by GregoryAlbouy**
## Description

Basic setup for a Go repo.

## Notes

- No conventions / contributing guidelines for now, as I suggest we use a dedicated repository for that
- Added some CI steps

### CODE REVIEW

**Review by moreirathomas — DISMISSED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      - yodastyleexpr
+
+    settings:
+      hugeParam:
+        sizeThreshold: 256
+      rangeValCopy:
+        sizeThreshold: 256
+
+  gofumpt:
+    lang-version: "1.16"
```
Comment (GregoryAlbouy) [inline .golangci.yml:107]:
```suggestion
    lang-version: "1.17"
```

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    local-prefixes: github.com/drykit-go/testx
+
+  misspell:
+    locale: US
+
+  revive:
+    enableAllRules: true
+
+  staticcheck:
+    go: "1.16"
```
Comment (GregoryAlbouy) [inline .golangci.yml:120]:
```suggestion
    go: "1.17"
```

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    checks: [all]
+
+  stylecheck:
+    go: "1.16"
```
Comment (GregoryAlbouy) [inline .golangci.yml:124]:
```suggestion
  go: "1.17"
```

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    checks: [all]
+
+linters:
+  disable-all: true
+  enable:
+    - bodyclose # enforce resp.Body.Close()
+    - deadcode
+    # - dupl # duplicate code
```
Comment (GregoryAlbouy) [inline .golangci.yml:132]:
```suggestion
   - dupl # duplicate code
```

File: .golangci.yml [hunk 5]
```diff
@@ -0,0 +1,158 @@
+run:
+  timeout: 1m
+  uniq-by-line: false
+
+linters-settings:
+  dupl:
+    threshold: 100
+
+  errcheck:
+    ignore: 'net/http:Write'
```
Comment (GregoryAlbouy) [inline .golangci.yml:10]: (deprecated)
```suggestion
    exclude-functions:
      - (net/http.ResponseWriter).Write
```

**Review by moreirathomas — APPROVED** [review]

**Pull request merged**